cmd/internal/flagparser/flag_parser.go

```diff
@@ -16,28 +18,60 @@ const DisableAll = "all"
 func NewFlagParser() *FlagParser {
 	fp := &FlagParser{}
 
-	flag.StringVar(&fp.Enable, "enable", EnableAll,
+	fp.flagSet = flag.CommandLine
+
+	fp.flagSet.StringVar(&fp.Enable, "enable", EnableAll,
+		`comma-separated list of enabled checkers`)
+	fp.flagSet.StringVar(&fp.ConfigFile, "config", "",
+		`name of JSON file containing checkers configurations`)
+	fp.flagSet.StringVar(&fp.Disable, "disable", "",
+		`comma-separated list of disabled checkers`)
+	fp.flagSet.BoolVar(&fp.WithExperimental, `withExperimental`, false,
+		`only for -enable=all, include experimental checks`)
+	fp.flagSet.BoolVar(&fp.WithOpinionated, `withOpinionated`, false,
+		`only for -enable=all, include very opinionated checks`)
+	fp.flagSet.IntVar(&fp.FailureExitCode, "failcode", 1,
+		`exit code to be used when lint issues are found`)
+	fp.flagSet.BoolVar(&fp.CheckGenerated, "checkGenerated", false,
+		`whether to check machine-generated files`)
+	fp.flagSet.BoolVar(&fp.ShorterErrLocation, "shorterErrLocation", true,
+		`whether to replace error location prefix with $GOROOT and $GOPATH`)
+
+	return fp
+}
+
+// FlagParser, where default values different from real ones, used in trick for define that
+// arguments was provided by user, or just was set up by default
+// See https://stackoverflow.com/a/51903637/4143494
+func newDefaultInvertedFlagParser() *FlagParser {
```

Comment:
this is a duplicate of `NewFlagParser`, maybe it will be better add a param to `NewFlagParser` which will enable this trick?
(and move comment closer to the trick 😉 )

Reply:
You mean do something like this?

```
func NewFlagParser(invertedDefaults bool) {
...
    if (invertedDefaults) {
          fp.flagSet.StringVar(&fp.Enable, "enable", "inverted",
		`comma-separated list of enabled checkers`)
    } else {
          fp.flagSet.StringVar(&fp.Enable, "enable", EnableAll,
		`comma-separated list of enabled checkers`)
    }
}
```

Or use the same with ternary workaround variations?

I thought about that before and come to conclusion that there is one disadvantage - mixing this trick logic with normal setup flags, and I don't see advantages of this. And it will really confuse developers. With "if" - will be copypaste again, just in one place, with ternary operator workarounds - will looks heavy.

But I found as I think best solution, invert defaults without hardcode flags, check please, commit https://github.com/go-critic/go-critic/pull/573/commits/9788a5c090a138e95890209e431754cd647fbad3